sinonjs/sinon
# Additions to 2152: fully tested runnable examples running in RunKit for stubs.md

#### Purpose (TL;DR) - mandatory
This is basically @andrewdroz's #2152 with some improvements:
- update all examples to use (a quite limited subset of) ES2015+ syntax
- make a test runner that verifies that all of the examples actually pass
- ~replace `@fatso83/mini-mocha` with Hapi's Lab test runner to avoid polluting `global`.~
- test the examples with the current version of Sinon by using `npm link`
- fix all prettier/eslint violations
- rename files to get usual editor support for javascript without any fiddling

 #### Solution  
See #2152 for more background info

 #### How to verify - mandatory
You can do the awkward excercise of building the docs to check the effect; commenting out stuff in `_config.yml`, building and then checkout out http://127.0.0.1:4000/release-source/release/stubs.

Or you can just have a look at the following image:
![2020-03-19_12-05_1](https://user-images.githubusercontent.com/618076/77064073-0e7e4080-69df-11ea-92af-8802d6e4b864.png)


You can also run the tests I state have been added by running `npm run test-runnable-examples ` 😄 

 #### Checklist for author
